docs/redis/redis.md

```diff
@@ -0,0 +1,360 @@
+
+Redis clients communicate with the Redis server using a protocol called
+**RESP** (REdis Serialization Protocol). The protocol is binary-safe and
+easy to be implement at client side, which was designed specifically for
+Redis.
+
+Redis server accepts commands composed of different arguments. Once a
+command is received, it's processed and a reply is sent back to the client.
+Two kind of Redis clients are widely used, the smart Redis client and
+simple Redis client. When using smart Redis client to connect the Redis
+cluster, it will send the commaonds to the right Redis server. Because the
+client will get the hash slots and server nodes mapping data at the bootrap
+time. When using simple Redis client,  the server address is needed, and
+all the requests are sent to this address, and the requests only send to
+the single Redis server.
+
+In Scylla terminology, the node receiving the request acts as the the
+coordinator, and often passes the request on to one or more other nodes,
+which hold copies of the requested data.  Any node of Scylla cluster can
+process the request correctly. In other words, Client can send request to
+any node of Redis cluster built on the top of Scylla. Obviously, the simple
+client can accesses the Redis cluster built on the top of Scylla, when
+provided address is not an IP but rather a single domain name, which a
+client resloves a random Scylla node IP address. The smart Redis client
+will query the mapping information between hash slots and Server nodes. We
+just fill the mapping with random Scylla node IP address.
+
+With Redis enabled on port 6379 (which is configurable), every Scylla node
+listens for Redis requests on this port. These requests, in
+[RESP](https://redis.io/topics/protocol) format over TCP, are parsed and
+result in calls to internal Scylla C++ functions.
+
+## 4. Data Model 
+
+Out of box, every Redis cluster supports 16 databases. The default database
+is 0 but it allows us to change that to any number from 0-15 (and allow us
+to configure Redis to support more databases). Each database provides
+a distinct keyspace, independent from the others. Use `SELECT n`
+to change databases. Redis allows us to store keys that map to any one of
+five different  basic data structure types:  STRINGs,  LISTs,  SETs,
+HASHes, and  ZSETs. (In fact, Now Redis has support other structure types,
+but the basic structure types are used widely). 
+
+In this proposal, We use the column family of Scylla to simulate the
+database within Redis. At the bootrap phase, the column familes with fixed
+name should be created (if not exists) or loaded. 
+
+We known that, Redis allows us to store keys that map to any one of the
+basic data structures. Scylla supports variant type, which allows us to
+define one table to store all kind of these Redis data. For example ,we
+create the table with the schame as follow:
+
+```
+CREATE TABLE redis （
+    key text,
+    value variant\<text, list\<text\>, map\<text\>\>,
```

Comment:
The map only has one item?

Reply:
No. This is one of the optional sulotion. As mentioned in the redis.md, this sulotion is not the not better one.